crates/iota-sdk-ffi/src/crypto/secp256r1.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use iota_crypto::{Signer, Verifier};
+use iota_types::SignatureScheme;
+
+use crate::{
+    error::SdkFfiError,
+    types::{
+        crypto::{Secp256r1PublicKey, Secp256r1Signature},
+        signature::{SimpleSignature, UserSignature},
+    },
+};
+
+#[derive(derive_more::From, uniffi::Object)]
+pub struct Secp256r1PrivateKey(pub iota_crypto::secp256r1::Secp256r1PrivateKey);
+
+#[uniffi::export]
+impl Secp256r1PrivateKey {
+    #[uniffi::constructor]
+    pub fn new(bytes: Vec<u8>) -> Self {
+        Self(iota_crypto::secp256r1::Secp256r1PrivateKey::new(
+            bytes.try_into().unwrap(),
+        ))
+    }
+
+    pub fn scheme(&self) -> SignatureScheme {
+        self.0.scheme()
+    }
+
+    /// Get the public key corresponding to this private key.
+    pub fn public_key(&self) -> Secp256r1PublicKey {
+        self.0.public_key().into()
+    }
+
+    /// Sign a message and return a Secp256r1Signature.
+    pub fn try_sign(&self, message: Vec<u8>) -> Secp256r1Signature {
+        <iota_crypto::secp256r1::Secp256r1PrivateKey as iota_crypto::Signer<
+            iota_types::Secp256r1Signature,
+        >>::try_sign(&self.0, &message)
+        .unwrap()
+        .into()
+    }
+
+    /// Sign a message and return a SimpleSignature.
+    pub fn try_sign_simple(&self, message: Vec<u8>) -> SimpleSignature {
+        <iota_crypto::secp256r1::Secp256r1PrivateKey as iota_crypto::Signer<
+            iota_types::SimpleSignature,
+        >>::try_sign(&self.0, &message)
+        .unwrap()
+        .into()
+    }
+
+    /// Sign a message and return a UserSignature.
+    pub fn try_sign_user(&self, message: Vec<u8>) -> UserSignature {
+        <iota_crypto::secp256r1::Secp256r1PrivateKey as iota_crypto::Signer<
+            iota_types::UserSignature,
+        >>::try_sign(&self.0, &message)
+        .unwrap()
+        .into()
+    }
+
+    pub fn verifying_key(&self) -> Secp256r1VerifyingKey {
+        Secp256r1VerifyingKey(self.0.verifying_key())
```

Comment:
```suggestion
        self.0.verifying_key().into()
```
more consistent with the rest